tests/components/template/test_light.py

```diff
@@ -378,6 +381,64 @@ async def test_on_action(hass, calls):
     assert len(calls) == 1
 
 
+async def test_on_action_with_transition(hass, calls):
+    """Test on action with transition."""
+    assert await setup.async_setup_component(
+        hass,
+        light.DOMAIN,
+        {
+            "light": {
+                "platform": "template",
+                "lights": {
+                    "test_template_light": {
+                        "value_template": "{{states.light.test_state.state}}",
+                        "turn_on": {
+                            "service": "test.automation",
+                            "data_template": {
+                                "transition": "{{transition}}",
+                            },
+                        },
+                        "turn_off": {
+                            "service": "light.turn_off",
+                            "entity_id": "light.test_state",
+                        },
+                        "supports_transition_template": "{{true}}",
+                        "set_level": {
+                            "service": "light.turn_on",
+                            "data_template": {
+                                "entity_id": "light.test_state",
+                                "brightness": "{{brightness}}",
+                                "transition": "{{transition}}",
+                            },
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    hass.states.async_set("light.test_state", STATE_OFF)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("light.test_template_light")
+    assert state.state == STATE_OFF
+
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: "light.test_template_light", ATTR_TRANSITION: 5},
+        blocking=True,
+    )
+
+    assert len(calls) == 1
+    print(calls[0].data)
```

Comment:
Leftover debug print.

Reply:
#51059 (+ a bunch more occurances)

Reply:
Hi, i can clean it up. Should i do this on my branch?

Reply:
@jacekpaszkowski I've already put it in #51059